src/gui/modals/FileDialog.cpp

```diff
@@ -87,22 +57,29 @@ FileDialog::FileDialog( QWidget *parent, const QString &caption,
 #endif
 
 #ifdef LMMS_BUILD_LINUX
+	QList<QUrl> m_addedSidebarUrls;
```

Comment:
You need to declare this variable in the header `FileDialog.h` for it to be remembered until later. (That's partly why the windows build fails atm)

You'll also need to define the destructor `FileDialog::~FileDialog` (it will be automatically called when the window is closed) and it should restore the sidebar to its original state by removing every `m_addedSideBarUrls` from it.